pymc/distributions/distribution.py

```diff
@@ -83,6 +85,60 @@
 PLATFORM = sys.platform
 
 
+def filter_shared_RNGs(params):
+    return [p for p in params if not isinstance(p, RandomGeneratorSharedVariable)]
+
+
+def rewrite_moment_scan_node(node):
+    if not isinstance(node.op, Scan):
+        return
+
+    node_inputs, node_outputs = node.op.inner_inputs, node.op.inner_outputs
+    op = node.op
+
+    local_fgraph_topo = io_toposort(node_inputs, node_outputs)
+
+    replace_with_moment = []
+    to_replace_set = set()
+
+    for nd in local_fgraph_topo:
+        if nd not in to_replace_set and isinstance(nd.op, (RandomVariable, SymbolicRandomVariable)):
+            replace_with_moment.append(nd.out)
+            to_replace_set.add(nd)
+    givens = {}
+    if len(replace_with_moment) > 0:
+        for item in replace_with_moment:
+            givens[item] = moment(item)
+    op_outs = clone_replace(node_outputs, replace=givens)
+
+    nwScan = Scan(
```

Comment:
Is there a way to check if there was actually any replacement done? If not, there's no point in creating a new Scan

Reply:
Yeah, I think we can add this check